zerver/openapi/zulip.yaml

```diff
@@ -18750,23 +18750,64 @@ paths:
+                    new value for who would have the permission to mention the
+                    group or only the new value for who would have the permission
+                    to mention the group.
+
+                    The setting value in the above mentioned cases can either be
+                    ID of a named user group that has permission to mention the
+                    group, or an object describing the set of users and groups
+                    who have permission mention the group.
+
+                    If the client passes the value as an object with `old` and
+                    `new` fields, and the `old` field value does not match the
+                    actual current value of who can mention the group, then the
+                    request will fail with error code EXPECTATION_MISMATCH and
+                    no changes will be applied.
+
+                    Clients with a user interface for editing this list should
+                    generally always pass this field as an object with `old`
+                    field included, giving the value the list had when the user
+                    started editing. Omitting this field can be appropriate when
+                    syncing the list from an external source of truth.
+
+                    This setting cannot be set to `"role:internet"` and `"role:owners"`
+                    system groups.
+
+                    **Changes**: In Zulip 9.0 (feature level 259), added support
+                    for clients to pass the setting value as an object containing
+                    both new and old vales of the setting.
+
+                    **Changes**: Before Zulip 9.0 (feature level 258), the
+                    `can_mention_group` this field was always an integer.
+
+                    **Changes**: Before Zulip 8.0 (feature level 198),
+                    the `can_mention_group` setting was named `can_mention_group_id`.
+
+                    New in Zulip 8.0 (feature level 191). Previously, groups
+                    could be mentioned if and only if they were not system groups.
+                  oneOf:
+                    - type: object
+                      additionalProperties: false
+                      properties:
+                        new:
+                          allOf:
+                            - description: |
+                                The new value for who would have the permission to
+                                mention the group.
+                            - $ref: "#/components/schemas/CanMentionGroup"
+                        old:
+                          allOf:
+                            - description: |
+                                The expected current value for who has the permission
+                                to mention the group.
+                            - $ref: "#/components/schemas/CanMentionGroup"
+                      required:
+                        - new
```

Comment:
This is how the allowed types are depicted currently. Think we would have to explain the types in description properly.
![Screenshot from 2024-05-10 23-48-22](https://github.com/zulip/zulip/assets/35494118/06331ddc-445c-4dd4-8afa-948c71bd3124)